parts/k8s/cloud-init/nodecustomdata.yml

```diff
@@ -6,135 +6,135 @@ write_files:
   encoding: gzip
   owner: root
   content: !!binary |
-    {{CloudInitData "provisionSource"}}
+    {{ CloudInitData "provisionSource" }}
 
 - path: /opt/azure/containers/provision.sh
   permissions: "0744"
   encoding: gzip
   owner: root
   content: !!binary |
-    {{CloudInitData "provisionScript"}}
+    {{ CloudInitData "provisionScript" }}
 
 - path: /opt/azure/containers/provision_installs.sh
   permissions: "0744"
   encoding: gzip
   owner: root
   content: !!binary |
-    {{CloudInitData "provisionInstalls"}}
+    {{ CloudInitData "provisionInstalls" }}
 
 - path: /opt/azure/containers/provision_configs.sh
   permissions: "0744"
   encoding: gzip
   owner: root
   content: !!binary |
-    {{CloudInitData "provisionConfigs"}}
+    {{ CloudInitData "provisionConfigs" }}
 
 - path: /opt/azure/containers/provision_cis.sh
   permissions: "0744"
   encoding: gzip
   owner: root
   content: !!binary |
-    {{CloudInitData "provisionCIS"}}
+    {{ CloudInitData "provisionCIS" }}
 
-{{if IsAzureStackCloud}}
+{{- if IsAzureStackCloud }}
```

Comment:
What's the purpose of this added `-` ?

Reply:
The `-` tells go templates to discard any whitespaces located before `{{`.

It's useful to discard empty line generated when a line only contains a go template statement.

https://play.golang.org/p/6YdJA8vOQhe

Reply:
Got it, thanks. Let's merge this in first:

https://github.com/Azure/aks-engine/pull/1684

That will probably cause merge conflicts with this changeset, FYI. So after we rebase and clear up conflicts, let's queue this up for E2E tests, review, and merge.

Thanks!